www/source/docs/developing-packages.html.md.erb

```diff
@@ -87,6 +87,47 @@ All plans must have a `plan.sh` or `plan.ps1` at the root of the plan context. T
 
 When writing a plan, it's important to understand that you are defining both how the package is built and the actions Chef Habitat will take when the Supervisor starts and manages the child processes in the package. The following sections explain what you need to do for each phase.
 
+## <a name="plan-targets" id="plan-targets" data-magellan-target="plan-targets" type="anchor">Writing a Plan for Multiple Platform Targets</a>
+
+You may want to create a plan for an application that can run on multiple platform targets. You can create target specific folders beneath either the root of your project or a top level `habitat` folder. Then save the plan, hooks, and configuration templates specific to a single platform all inside of that target specific folder.
+
+For example, an application targeting linux, Linux kernel 2, and Windows may have the following structure:
+
+```
+app_root/
+├── x_86_64-linux/
+|   |   plan.sh
+|   └── hooks/
+|           run
+├── x_86_64-linux-kernel2/
+|   |   plan.sh
+|   └── hooks/
+|           run
+└── x_86-64-windows/
+    |   plan.ps1
+    └── hooks/
+            run
+```
+
+The build script will look for the base of your plan in the following locations:
+
+- `<app_root>/`
+- `<app_root>/habitat/`
+- `<app_root>/<target>/`
+- `<app_root>/habitat/<target>/`
+
+If it finds a plan both inside as well as outside of a target folder, the target specific plan will be used as long as that is the target currently being built.
```

Comment:
This seems like something we'd want to warn about (at the very least); I think it might justify stopping the build, though (assuming that doesn't break a bunch of existing plans).

Having a mix of locations for a single package seems like a recipe for confusion and disaster.

Is there a reason we allow `<app_root>/<target>` in the first place? It seems that once you have to specify a target, you've got more than one, by definition, at which point _requiring_ them to all be contained in `<app_root>/habitat` makes a lot of sense. I suppose if you're just dealing with a pure plan repository, and not code + plans, target directories in the root makes sense. Perhaps drawing that distinction here in the documentation would be good; we should be encouraging people towards the patterns they should be using, and saying why you'd want to use one over the other, rather than just saying "you can do X, Y, or Z; have fun!"

Reply:
>Is there a reason we allow <app_root>/<target> 
>I suppose if you're just dealing with a pure plan repository

That's essentially it, core plans don't use the `habitat` directory, so for the time being we'd need to support that case.  We could make that change, but I think that warrants a larger discussion than this pr around how we manage changes to Plans (the entire structure of plan.sh/ps1, hooks, variants, etc.) 

:+1: on encouraging people to use the appropriate pattern for their use case though.

Reply:
> This seems like something we'd want to warn about (at the very least); I think it might justify stopping the build, though (assuming that doesn't break a bunch of existing plans).

> Having a mix of locations for a single package seems like a recipe for confusion and disaster.

I agree a warning would be good here and will add that. We don't fail because that would break some core plans (like glibc as one example). However after the refresh is complete, it may make sense to correct those plans and then adjust here.